cluster/manifests/serviceaccounts_rbac.yaml

```diff
@@ -0,0 +1,123 @@
+kind: ClusterRole
+metadata:
+  name: jobs-operator-role-cluster
+rules:
+  # Framework: knowing which other operators are running (i.e. peering).
+  - apiGroups: [kopf.dev]
+    resources: [clusterkopfpeerings]
+    verbs: [list, watch, patch, get]
+
+  # Framework: runtime observation of namespaces & CRDs (addition/deletion).
+  - apiGroups: [apiextensions.k8s.io]
+    resources: [customresourcedefinitions]
+    verbs: [list, watch]
+  - apiGroups: [""]
+    resources: [namespaces, pods, jobs, services]
+    verbs: [list, watch, create]
+  - apiGroups: [batch]
+    resources: [jobs]
+    verbs: [list, watch, create]
+  - apiGroups: [apps]
+    resources: [deployments]
+    verbs: [list, watch, create]
+  # Framework: admission webhook configuration management.
+  - apiGroups: [admissionregistration.k8s.io/v1, admissionregistration.k8s.io/v1beta1]
+    resources: [validatingwebhookconfigurations, mutatingwebhookconfigurations]
+    verbs: [create, patch]
+
+  # Application: read-only access for watching cluster-wide.
+  - apiGroups: [kopf.dev]
+    resources: [kopfexamples]
+    verbs: [list, watch]
+  - apiGroups: ["curibio.dev"]
+    resources: [jobrunners]
+    verbs: [list, watch, patch, get]
+  - apiGroups: [""]
+    resources: [events]
+    verbs: [create]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  namespace: kopf
+  name: jobs-operator-role-namespaced
+rules:
+  # Framework: knowing which other operators are running (i.e. peering).
+  - apiGroups: [kopf.dev]
+    resources: [kopfpeerings]
+    verbs: [list, watch, patch, get]
+
+  # Framework: posting the events about the handlers progress/errors.
+  - apiGroups: [""]
+    resources: [events]
+    verbs: [create]
+
+  # Application: watching & handling for the custom resource we declare.
+  - apiGroups: [kopf.dev]
+    resources: [kopfexamples]
+    verbs: [list, watch, patch]
+
+  # Application: other resources it produces and manipulates.
+  # Here, we create Jobs+PVCs+Pods, but we do not patch/update/delete them ever.
+  - apiGroups: [batch, extensions]
+    resources: [jobs]
+    verbs: [create, list]
+  - apiGroups: [""]
+    resources: [pods, persistentvolumeclaims]
+    verbs: [create, list]
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: jobs-operator-role-cluster
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: jobs-operator-role-cluster
+subjects:
+  - kind: ServiceAccount
+    name: jobs-operator
+    namespace: kopf
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: jobs-operator-role-namespaced
+  namespace: kopf
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: jobs-operator-role-namespaced
+subjects:
+  - kind: ServiceAccount
+    name: jobs-operator
+    namespace: kopf
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  namespace: pulse3d
+  name: pulse3d-role-namespaced
```

Comment:
what's this for?

Reply:
@jperson  the queue-processor pod is started in the pulse3d namespace, it didn't have the correct permissions to then spin up the jobs

Reply:
got it, so each queue-processor will get one of these for the namespace it runs in

Reply:
Yes!